From 46e0af2942e23fb3cf1c313e58e4081877d4f289 Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Thu, 30 Aug 2012 18:57:06 +0000
Subject: ip6: ignore cached/cloned routes when retrieving IPv6 config (lp:1038541) (bgo #671767)

Should also ignore them when building up the initial IPv6 config.
---
Index: network-manager/src/ip6-manager/nm-ip6-manager.c
===================================================================
--- network-manager.orig/src/ip6-manager/nm-ip6-manager.c	2012-08-31 09:56:03.000000000 +0200
+++ network-manager/src/ip6-manager/nm-ip6-manager.c	2012-08-31 09:56:03.000000000 +0200
@@ -1284,6 +1284,12 @@
 		if (rtnl_route_get_family (rtnlroute) != AF_INET6)
 			continue;
 
+		/* And ignore cache/cloned routes as they aren't part of the interface's
+		 * permanent routing configuration.
+		 */
+		if (rtnl_route_get_flags (rtnlroute) & RTM_F_CLONED)
+			continue;
+
 		nldest = rtnl_route_get_dst (rtnlroute);
 		if (!nldest || nl_addr_get_family (nldest) != AF_INET6)
 			continue;
